vmdb/app/models/service_template_orchestration.rb

```diff
@@ -1,6 +1,16 @@
 class ServiceTemplateOrchestration < ServiceTemplate
   include ServiceOrchestrationMixin
 
+  before_save :remove_invalid_resource
+
+  def remove_invalid_resource
+    service_resources.each { |r| r.delete unless r.resource(true) }
```

Comment:
I am not deleting an instance in a collection but delete a record in the table. `true` argument is required because the resource may be already removed from table but the memory still holds the object. This is identified by spec tests.

I can use `delete_if` to remove the resource from memory also.